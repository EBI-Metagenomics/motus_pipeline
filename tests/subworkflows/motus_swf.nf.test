nextflow_process {

    name "Test SubWorkflow motus with db download"
    script "subworkflows/motus_swf.nf"
    process "MOTUS_SUBWF"

    test("motus with db download") {

        when {
            params {
                reads = "${baseDir}/tests/subworkflows/fixtures/motus/test.fastq.gz"
            }
            process {
                """
                input[0] = channel.fromPath(params.reads)
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }
}
