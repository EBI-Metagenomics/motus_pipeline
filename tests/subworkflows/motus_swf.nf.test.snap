{
    "motus with db download": {
        "content": [
            {
                "motus_tsv": [
                    "test.fastq.tsv:md5,374013e14d9130e1b6fd919ae6b507e7"
                ]
            }
        ],
        "timestamp": "2023-03-23T11:08:53+0000"
    }
}